buildSrc/src/main/groovy/BuildkitBuild.groovy

```diff
@@ -0,0 +1,357 @@
+    @Input
+    final Property<Boolean> isBuildx =
+            getObjects().property(Boolean).convention(true)
+
+    // Property for loading images into docker
+    @Input
+    final Property<Boolean> useDocker =
+            getObjects().property(Boolean).convention(false)
+
+    // The images contain shortened filenames if true
+    @Input
+    final Property<Boolean> useShortName =
+            getObjects().property(Boolean).convention(false)
+
+    // Property to set the output image's repository
+    @Input
+    final Property<String> containerRepo =
+            getObjects().property(String).convention('')
+
+    // Property used to set custom image tag
+    @Input
+    final Property<String> containerTag =
+            getObjects().property(String).convention('')
+
+    // Property used to set custom image name
+    @Input
+    final Property<String> containerName =
+            getObjects().property(String).convention('')
+
+    // Property to set the base image's name
+    @Input
+    final Property<String> baseImageName =
+            getObjects().property(String).convention('azul/zulu-openjdk')
+    // Property to set the base image's tag
+    @Input
+    final Property<String> baseImageTag =
+            getObjects().property(String).convention('11')
+
+    // Arguments passed to the image entrypoint
+    @Input
+    final ListProperty<String> arguments =
+            getObjects().listProperty(String)
+
+    // Tasklist to get the sourcefiles from
+    @Input
+    final ListProperty<Task> sourceTasks =
+            getObjects().listProperty(Task)
+
+    // Files generated in the build that need to be inside the container
+    @PathSensitive(RELATIVE)
+    @InputFiles
+    final ConfigurableFileCollection extraSourceFiles =
+            getObjects().fileCollection()
+
+    // JDBC driver files generated in the build that need to be inside the container
+    @PathSensitive(RELATIVE)
+    @InputFiles
+    final ConfigurableFileCollection jdbcDriverFiles =
+            getObjects().fileCollection()
+
+    // Used to create a folder for plugins in tools:plugins task
+    @Input
+    final Property<String> subDir =
+            getObjects().property(String).convention('')
+
+    // Overrides name of the jar file being executed at the entrypoint
+    @Input
+    final Property<String> overrideEntryName =
+            getObjects().property(String).convention('')
+
+    // Publishing to dockerhub
+    @Input
+    final Property<Boolean> dockerHubPublish =
+            getObjects().property(Boolean).convention(false)
+
+    BuildkitBuild() {
+        group = 'publishing'
+
+        if (!isBuildx) {
+            try {
+                (new Socket('127.0.0.1', 3476)).close();
+                logger.quiet("Buildkit daemon found")
+            }
+            catch (SocketException e) {
+                throw new GradleException("No daemon found. Please connect to available buildkit daemon (and port forward it to 3476) and start again")
```

Comment:
Could we please amend `README.md` explaining how to stand-up BuildKit Daemon?

Reply:
readme added!